### PR TITLE
qt5: 5.14.2 -> 5.15.0 

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

Update to Qt 5.15.

This is just a rebased version of https://github.com/petabyteboy/nixpkgs/pull/1
All thanks go to @nrdxp and @petabyteboy who did the actual work

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by petabyteboy** [comment]
I have rebased this to current master and added the change from https://github.com/NixOS/nixpkgs/pull/95772 to Qt 5.15 as well.

**Comment by petabyteboy** [comment]
This should probably go to the staging branch, can you change that?

**Comment by gebner** [comment]
I've rebased it to staging.

**Comment by ttuegel** [comment]
I moved the milestone. I want to have this in NixOS 20.09 because the support window for Qt 5.14 will end before NixOS 21.03. I'm building this locally right now, but obviously I won't be able to finish reviewing until tomorrow. :sweat_smile:

File: pkgs/development/libraries/qt-5/modules/qtbase.nix [hunk 1]
```diff
@@ -312,7 +311,6 @@ stdenv.mkDerivation {
         [
           "-${lib.optionalString (compareVersion "5.9.0" < 0) "no-"}rpath"
 
-          "-system-xcb"
```
Comment (ttuegel) [inline pkgs/development/libraries/qt-5/modules/qtbase.nix:315]: Have these configuration changes been tested with Qt 5.12 and Qt 5.14?

**Comment by ttuegel** [comment]
I still want to merge this, but we will _not_ make Qt 5.15 the default version. It is not officially supported by many packages yet, particularly it is unsupported by Plasma.

**Comment by worldofpeace** [comment]
@ttuegel So we have to use a version of qt in stable for plasma that won't be supported?

**Comment by ajs124** [comment]
From looking at all-packages.nix, default for linux is 5.14 right now, which will be supported until 2020-12-12.
5.12 is still supported until 2020-12-12.

**Comment by ttuegel** [comment]
> So we have to use a version of qt in stable for plasma that won't be supported (at some tiny point before 21.03)?

@worldofpeace We actually need to revert to Qt 5.12 in stable. Plasma 5.18 doesn't support Qt 5.14, and Plasma 5.19 is only supported until October (and isn't merged anyway).

**Comment by gebner** [comment]
> @worldofpeace We actually need to revert to Qt 5.12 in stable. Plasma 5.18 doesn't support Qt 5.14, and Plasma 5.19 is only supported until October (and isn't merged anyway).

I don't think we should stick with LTS versions of Plasma if that means we can't upgrade Qt.  AFAICT [Plasma LTS releases only come around once every 1.5 years](https://community.kde.org/Schedules/Plasma_5), so we would only be able to upgrade Qt every 1.5 years either.

If maintaining non-LTS versions (5.19) of Plasma is infeasible, could we instead override the Qt version only for Plasma and keep the system version at 5.15?

BTW, did you run into any other build errors with 5.15 aside from Plasma?

**Comment by ttuegel** [comment]
> I don't think we should stick with LTS versions of Plasma if that means we can't upgrade Qt.

Plasma 5.18 is the only version of Plasma that will have support through our support window for NixOS 20.09, so I don't think we have much choice there. Plasma 5.20 won't be released until October, and Plasma 5.19's support window expires shortly after. We don't have the resources to be backporting security patches onto Plasma 5.19 for six months. Qt 5.12 is also an LTS release, with security updates beyond the support window of NixOS 20.09.

The status of different version combinations is:

1. I can't even build Qt 5.15 right now, let alone build anything with it.
2. Plasma 5.18 officially supports Qt 5.12. Plasma 5.18 builds with Qt 5.14, but it's not officially supported upstream and it's very glitchy. (I'm using it right now, but I haven't been able to update my main work machine because it's _so bad_.)
3. Plasma 5.19 supports Qt 5.14 officially. It does not support Qt 5.15, so I expect that combination to be buggy. The support window ends when Plasma 5.20 is released in October. It's too late to update for NixOS 20.09 now, and even if I begged for an exception (which I won't) I can't see how we can support it until NixOS 21.03.

I can't _imagine_ foisting any combination other than Plasma 5.18 + Qt 5.12 on users of NixOS 20.09. The combination is supported upstream and each has LTS support through our support window. I think anything else would be an utter disaster.

**Comment by worldofpeace** [comment]
@ttuegel All of that sounds good to me.  I encourage any option that makes Plasma use Qt 5.12.

**Comment by ttuegel** [comment]
I'm going to open two separate pull requests:

1. Add Qt 5.15 and make it the default for everything except Plasma and KDE Applications, which stay at Qt 5.14. (I know, yet another pull request, I'm sorry.) I think this one can go directly into `master` because it shouldn't be a mass rebuild.
2. Revert Plasma 5.18 to Qt 5.12. This needs to go into `staging` because it is a mass rebuild. This does _not_ need to hold up the release branch because we can always push it into the release staging branch after. (It's a 3-line change, there won't be conflicts.)

I think this meets everyone's needs, and we can get it done quickly.

**Pull request closed without merging**